mne/encoding/feature.py

```diff
@@ -0,0 +1,341 @@
+"""A collection of classes for transforming, preprocessing, and fitting."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn, _reject_data_segments
+
+
+class DataDelayer(object):
```

Comment:
I'm a bit lost on this object, but it seems to delay X, not y (but since it's an encoding model, I'm not sure whether X is supposed to be the recordings or the conditions...)